### ISSUE TITLE
python test fails

### ISSUE THREAD

**Issue by yarikoptic** [openmeeg/openmeeg#101]
Trying to build openmeeg on my laptop (to possibly avoid having its elderly package removed from debian, are you compatible with vtk6 aren't you? ;)) but `openmeeg_python` test fails.  Below I have added a print out of actual difference and of v and m (although not quite useful) so line number is offset by 3:

```
249: 
249: assemble SurfSourceMat with 42 mesh_source located in domain "Brain".
249: 
249: OPERATOR N ... (arg : mesh 1 , mesh  )
249: OPERATOR D(Optimized) ... (arg : mesh 1 , mesh  )
249: gain_meg_surf_one_line : 162 x 42
249: est_meg    : 162 x 6
249: est_meg_adjoint    : 162 x 6
249: est_eeg    : 32 x 6
249: est_eeg_adjoint    : 32 x 6
249: 0.0127881934591
249: <openmeeg.Vector; proxy of <Swig Object of type 'OpenMEEG::Vector *' at 0x7f11a360c840> >
249: <openmeeg.Matrix; proxy of <Swig Object of type 'OpenMEEG::Matrix *' at 0x7f11a360c810> >
249: Traceback (most recent call last):
249:   File "/home/yoh/deb/gits/pkg-exppsy/openmeeg/obj-x86_64-linux-gnu/OpenMEEG/build/Wrapping/src/test_python.py", line 166, in <module>
249:     assert((v-m.getcol(0)).norm() < 1e-15)
249: AssertionError
1/1 Test #249: openmeeg_python ..................***Failed    2.43 sec
```

I am on debian amd64 testing/sid.  Let me know if any other specifics needed.

**Comment by agramfort** [comment]
I have not used the python bindings in a while.

as I have no time to look into it now, can you just disable python for this version?

**Comment by papadop** [comment]
Yes in theory, we support vtk6 (and kept support for vtk5) as far as I know....
For the python failure, I also saw it. If I remember well, this is nothing serious, just a minor precision difference in matrix operations.... So the comparison threshold needs to be changed (again if I'm not wrong). I'll try to have a look on correcting that (and other things) at the end of the week. Sorry, very busy these last days.

**Comment by yarikoptic** [comment]
ok -- for my 'experimental' build I have just skipped this test, but I think I will just raise threshold to e.g. 0.1 ... I guess for the test difference should be "normed" by e.g. the mean of the compared matrices -- abs values difference is hard for me to appreciate since I have no clue in the nature of those ;)

**Comment by papadop** [comment]
OK. I have a fix for the problem. Actually, it was a bug that was introduced when I upgraded to the numpy 1.6 (or 1.7 ?) API. The pull request is coming.

**Comment by papadop** [comment]
But I uncovered another problem which is related to the ownership of objects between C++ and python. Consider the following code:

v=hm(1,10,1,1).getcol(0)
vec = om.asarray(v)
print(vec)

This works correctly. But the "equivalent" code:

m = hm(1,10,1,1)
print(om.asarray(m.getcol(0)))

does not. The reason ? getcol() does a copy and returns a temporary object. asarray does not copy the data and does not inform the C++ that the object is referenced (openmeeg vectors use reference counting). So right after the asarray function exits, the temp is destroyed and the python array references freed memory. The only solution I see would be to increment the vector reference counter in the asarray function and then make the python the owner of the memory (which means that we need to tag the getcol and similar methods with the swig %newobject to avoid creating memory leaks. Not sur this works in all situations thought.... Other ideas ??

I do not understand exactly why it does work when the python object is stored (1st example). Maybe, this is just by chance and the fix above will work correctly. If however, there is some swig magic, then incrementing the ref counter in asarray would create a memory leak in that case. I need to investigate more.

**Comment by papadop** [comment]
PR #109
